tests/specs/spa/capture-interactions.e2e.js

```diff
@@ -61,4 +63,74 @@ describe('spa captures interaction when', () => {
       ]
     }))
   })
+
+  it.withBrowsersMatching(notIE)('hashchange is followed by a popstate', async () => {
+    const url = await browser.testHandle.assetURL('instrumented.html')
+    await Promise.all([
+      browser.testHandle.expectInteractionEvents(),
+      browser.url(url).then(() => browser.waitForAgentLoad())
+    ])
+    const hashFragment = 'otherurl'
+    await Promise.all([
+      browser.testHandle.expectInteractionEvents(), // discard first hashchange interaction
+      browser.execute(function (hashFragment) {
+        window.location.hash = hashFragment
+      }, hashFragment)
+    ])
+
+    const [popstateIxnPayload] = await Promise.all([
+      browser.testHandle.expectInteractionEvents(),
+      browser.execute(function () {
+        window.addEventListener('popstate', function () { setTimeout(newrelic.interaction().createTracer('onPopstate')) })
+        window.history.back()
+      })
+    ])
+
+    const parsedUrl = new URL(url)
+    expect(popstateIxnPayload.request.body[0]).toEqual(expect.objectContaining({
+      category: 'Route change',
+      type: 'interaction',
+      trigger: 'popstate',
+      oldURL: parsedUrl.origin + parsedUrl.pathname + '#' + hashFragment,
+      newURL: parsedUrl.origin + parsedUrl.pathname, // should be the original asset url
+      children: [
+        expect.objectContaining({
+          type: 'customTracer',
+          name: 'onPopstate',
+          children: []
+        })
+      ]
+    }))
+  })
+
+  it('hashchange in 2nd event listener', async () => {
```

Comment:
I think this name could be a little more clear. I am assuming that the purpose of this test is that the interaction is counting the window addEventListener callback in the interaction created by the click???

Reply:
```suggestion
  it('Spa does not prevent the bubbling of events', async () => {
```